src/go/configgenerator/filterconfig/filter_generator.go

```diff
@@ -266,6 +275,32 @@ func updateProtoDescriptor(service *confpb.Service, apiNames []string, descripto
 	return newData, nil
 }
 
+func preserveDefaultHttpBinding(httpRule *ahpb.HttpRule, defaultPath string) {
+	defaultBinding := &ahpb.HttpRule{Pattern: &ahpb.HttpRule_Post{defaultPath}, Body: "*"}
+
+	// Check existence of the default binding in httpRule's additional_bindings to avoid duplication.
+	defaultBindingExisted := false
+	for _, addtionalBinding := range httpRule.AdditionalBindings {
+		if proto.Equal(addtionalBinding, defaultBinding) {
+			defaultBindingExisted = true
+			break
```

Comment:
`return`

No need for variable `defaultBindingExisted`. Just return early if you find it, no need to run the rest of the function.